web-ui/components/Games/Maze.vue

```diff
@@ -13,159 +13,47 @@
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-          ],
-          [
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-          ],
-          [
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-          ],
-          [
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-          ],
-          [
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-          ],
-          [
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-          ],
-          [
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-          ],
-          [
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-          ],
-          [
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Open",
-          ],
-        ],
-      };
-    },
-    mapMaze() {
-      const map = this.gameState.map;
-      const [playerX, playerY] = [
-        this.gameState.player.x,
-        this.gameState.player.y,
-      ];
-      const [exitX, exitY] = [this.gameState.exit.x, this.gameState.exit.y];
+      },
+      gameState: {
+        visualMap: this.emptyMaze(),
+      },
+      emptyMap: this.emptyMaze(),
+    };
+  },
+  updated() {
+    this.getMaze();
+  },
+  methods: {
+    async getMaze() {
+      const data = (await this.fetchMaze()).data;
+      const map = await data.map;
+      const [playerX, playerY] = await [data.player.x, data.player.y];
+      const [exitX, exitY] = await [data.exit.x, data.exit.y];
```

Comment:
You don't need await on everything - only things that return promises. These three are all already just data since you have already awaited on the promise returned by `this.fetchMaze()`.